### PR TITLE
Move table column filtering to modal

### PR DESCRIPTION
**Pull request by jake-bassett**

### CODE REVIEW

File: projects/components/src/table/header/table-header-cell-renderer.component.ts [hunk 1]
```diff
@@ -18,25 +25,43 @@ import { TableColumnConfigExtended } from '../table.service';
...
+          <ht-popover-trigger>
+            <div #trigger>
+              <ht-icon icon="${IconType.MoreHorizontal}" size="${IconSize.Small}"></ht-icon>
+            </div>
+          </ht-popover-trigger>
+          <ht-popover-content>
+            <div [style.min-width.px]="trigger.offsetWidth" class="popover-content">
+              <div class="popover-item" (click)="this.onFilterValues()" *ngIf="this.isFilterable">Filter Values</div>
+              <div class="popover-item-divider" *ngIf="this.isFilterable"></div>
+              <div class="popover-item" (click)="this.onSortChange(SORT_ASC)">
```
Comment (aaron-steinfeld) [inline projects/components/src/table/header/table-header-cell-renderer.component.ts:46]: Are these alternate ways of sorting? Also, our sort is a tri-state right now, but this only shows two options (one of which may already be applied) with no way of unsetting the sorting.

File: projects/components/src/filtering/filter-modal/in-filter-modal.component.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      .buildFilter(this.modalData.attribute, FilterOperator.In, [...this.selected.values()].sort(sortUnknown));
+
+    if (this.selected.size > 0) {
+      this.filterUrlService.addUrlFilter(this.modalData.metadata, filter);
+    } else {
+      const incompleteFilter: IncompleteFilter = {
+        ...filter,
+        value: undefined
+      };
+      this.filterUrlService.removeUrlFilter(this.modalData.metadata, incompleteFilter);
```
Comment (aaron-steinfeld) [inline projects/components/src/filtering/filter-modal/in-filter-modal.component.ts:84]: I'd expect us to close the modal after applying

File: projects/components/src/table/header/table-header-cell-renderer.component.ts (hunk 1 above)
Comment (jake-bassett) [reply projects/components/src/table/header/table-header-cell-renderer.component.ts:46]: Correct. These are alternatives. Clicking on the header cycle sorting as usual.

Comment (aaron-steinfeld) [reply projects/components/src/table/header/table-header-cell-renderer.component.ts:46]: Should there be a third option to remove sort to match the tristate?

**Review by aaron-steinfeld — APPROVED** [review]

**Review by aaron-steinfeld — APPROVED** [review]

**Review by aaron-steinfeld — APPROVED** [review]

**Pull request merged**